pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -382,21 +386,30 @@ type derivedInput struct {
 	Values      map[string]any
 }
 
-func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]string, error) {
+func (d *DeploymentController) render(templateName string, mi TemplateInput, deployment *appsv1.Deployment) ([]string, error) {
 	cfg := d.injectConfig()
 
 	template := cfg.Templates[templateName]
 	if template == nil {
 		return nil, fmt.Errorf("no %q template defined", templateName)
 	}
+
+	// istio.io/gateway-name is the only label that allows to attach ProxyConfig to a Gateway
+	workloadLabels := map[string]string{"istio.io/gateway-name": mi.Name}
+	podAnnotations := map[string]string{}
+	if deployment != nil && deployment.Spec.Template.Annotations != nil {
+		podAnnotations = deployment.Spec.Template.Annotations
+	}
+
```

Comment:
Can you add a few lines of comments on what this does  (to the code) ?  I would guess it takes ProxyConfig CR, falls back to the annotation and finally MeshConfig ?